src/app-router.js

```diff
@@ -22,14 +23,7 @@ export class AppRouter extends Router {
 
   loadUrl(url) {
     return this.createNavigationInstruction(url)
-      .then(instruction => this.queueInstruction(instruction))
-      .catch(error => {
```

Comment:
Sounds good. If you can make that one fix, I think it's good to go. I like the direction you are going with all this.